packages/openactive-integration-tests/test/features/payment/simple-book-with-payment/implemented/b-with-incomplete-payment-details-test.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable no-unused-vars */
+const chai = require('chai');
+const chakram = require('chakram');
+const { RequestState } = require('../../../../helpers/request-state');
+const { FlowHelper } = require('../../../../helpers/flow-helper');
+const { FeatureHelper } = require('../../../../helpers/feature-helper');
+const sharedValidationTests = require('../../../../shared-behaviours/validation');
+const { GetMatch, C1, C2, B } = require('../../../../shared-behaviours');
+
+const { expect } = chakram;
+/* eslint-enable no-unused-vars */
+
+FeatureHelper.describeFeature(module, {
+  testCategory: 'payment',
+  testFeature: 'simple-book-with-payment',
+  testFeatureImplemented: true,
+  testIdentifier: 'b-with-incomplete-payment-details',
+  testName: 'IncompletePaymentDetailsError must be returned in the case that payment details are not supplied',
+  testDescription: 'An usuccessful end to end booking, because identifier is missing in payment property in B request.',
+  // The primary opportunity criteria to use for the primary OrderItem under test
+  testOpportunityCriteria: 'TestOpportunityBookablePaid',
+  // The secondary opportunity criteria to use for multiple OrderItem tests
+  controlOpportunityCriteria: 'TestOpportunityBookable',
+},
+function (configuration, orderItemCriteria, featureIsImplemented, logger) {
+  describe('Missing identifier in payment property at B', () => {
+    const state = new RequestState(logger, { bReqTemplateRef: 'incorrectOrderDueToMissingIdentifierInPaymentProperty' });
+    const flow = new FlowHelper(state);
+
+    beforeAll(async function () {
+      await state.fetchOpportunities(orderItemCriteria);
+      return chakram.wait();
+    });
+
+    afterAll(async function () {
+      await state.deleteOrder();
+      return chakram.wait();
+    });
+
+    describe('Get Opportunity Feed Items', () => {
+      (new GetMatch({
+        state, flow, logger, orderItemCriteria,
+      }))
+        .beforeSetup()
+        .successChecks()
+        .validationTests();
+    });
+
+    describe('C1', () => {
+      (new C1({
+        state, flow, logger,
+      }))
+        .beforeSetup()
+        .successChecks()
+        .validationTests();
+    });
+
+    describe('C2', function () {
+      (new C2({
+        state, flow, logger,
+      }))
+        .beforeSetup()
+        .successChecks()
+        .validationTests();
+    });
+
+    describe('B', function () {
+      (new B({
+        state, flow, logger,
+      }))
+        .beforeSetup()
+        .itResponseReceived()
+        .validationTests();
+
+      it('should return 400, with an IncompletePaymentDetailsError error', () => {
+        chai.expect(state.bResponse.response.statusCode).to.equal(400);
+        chai.expect(state.bResponse.body).to.have.property('@type', 'IncompletePaymentDetailsError');
```

Comment:
This should instead use the library function: 
https://github.com/openactive/openactive-test-suite/blob/fd83edb5aeab79f278e3bda35f163ab58b843f4d/packages/openactive-integration-tests/test/shared-behaviours/errors.js#L14